### PR TITLE
doc: exactly how to build GD in local machine

### PR DESCRIPTION
**Pull request by shenlebantongying**
Replace https://github.com/xiaoyifang/goldendict-ng/pull/1663
close https://github.com/xiaoyifang/goldendict-ng/issues/1662

### CODE REVIEW

**Comment by shenlebantongying** [comment]
I think this is good enough.

File: website/docs/developer.md [hunk 1]
```diff
@@ -1,34 +1,110 @@
...
+
+Then, hit the "Run" button at bottom-right corner should build and run GoldenDict.
+
+### Command Line only
+See [how to build from source](howto/build_from_source.md).
+
+[Qt's doc: Building projects on the command line](https://doc.qt.io/qt-6/cmake-build-on-cmdline.html)
+
+### Visual Studio
+VS2022 has CMake support. After opening the source code folder, VS will starts to configure CMake immediately an failed immediately.
```
Comment (xiaoyifang) [inline website/docs/developer.md:69]: > VS will starts to configure CMake immediately an failed immediately.

File: website/docs/developer.md [hunk 2]
```diff
@@ -1,34 +1,110 @@
...
+![](https://github.com/xiaoyifang/goldendict-ng/assets/20123683/bd87155e-2e61-41d5-81e2-7bfb1f13c4c4)
+
+### CLion
+
+### XCode
+
+
+### CMake GUI
+
+? I have no idea how to make it work.
```
Comment (xiaoyifang) [inline website/docs/developer.md:102]: Leave it blank

File: website/docs/developer.md [hunk 3]
```diff
@@ -1,34 +1,110 @@
...
 
-Please follow [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) and write modern C++ code.
+### LSP + Editor?
```
Comment (xiaoyifang) [inline website/docs/developer.md:104]: original line is better

File: website/docs/developer.md [hunk 4]
```diff
@@ -1,34 +1,104 @@
...
+
+Then, hit the "Run" button at bottom-right corner should build and run GoldenDict.
+
+### Command Line only
+See [how to build from source](howto/build_from_source.md).
+
+[Qt's doc: Building projects on the command line](https://doc.qt.io/qt-6/cmake-build-on-cmdline.html)
+
+### Visual Studio
+VS2022 has CMake support. After opening the source code folder, VS will starts to configure CMake immediately and fail immediately.
```
Comment (xiaoyifang) [inline website/docs/developer.md:69]: I think `and fail immediately`   should be removed.

**Pull request merged**